python/lsst/pipe/tasks/skyCorrection.py

```diff
@@ -178,6 +177,11 @@ class SkyCorrectionConfig(PipelineTaskConfig, pipelineConnections=SkyCorrectionC
         dtype=FocalPlaneBackgroundConfig,
         doc="Initial background model, prior to sky frame subtraction",
     )
+    doBgModel1 = Field(
+        dtype=bool,
+        default=True,
+        doc="If False, adds back initial background model after sky",
+    )
```

Comment:
I would change the name of this config parameter. I see the symmetry in making it similar to doBgModel2 but in that case the config option chooses whether or not to do the final cleanup and background subtraction. Here, it looks like the initial background subtraction is done in order to do the sky subtraction, then it is added back in if `doBgModel1 == False`. So I would recommend calling this `restoreOriginalBackground` or something similar that doesn't make it sound as if the model 1 background subtraction is skipped completely.

Reply:
How about `restoreBgModel1`?  To keep the symmetry but clarify the usage.  And then the default value will swap to `False`.

Reply:
Maybe I misunderstand the code, but it isn't restoring bgModel1, right? I thought that it's restoring the background before bgModel1 was made and subtracted. Maybe `undoBgModel1` or `removeBgModel1` or `revertBgModel1`?

Reply:
The order of operations facilitated here is:

1. subtract bgModel1
2. subtract sky
3. remove bgModel1 from the background list
4. subtract bgModel2

At the end of this, the `skyCorr` output background list should consist of:

1. the inverted original detector-level background elements (4 of them)
2. no bgModel1 (this has been used to fit the sky frame, and then removed from the bg list)
3. sky frame
4. bgModel2

With that in mind, I do agree that naming this config `doBgModel1` makes it seem like a similar operation to `doBgModel2`, but it does act differently. I wasn't sure which alternative to endorse; I realize that in describing the order of operations and dataset outputs above I twice referred to this as a removal. However, out of context, "remove BG Model 1" sounds like you're subtracting the model, which is the opposite of what we're doing here.

So with that said, I think `undoBgModel1` is a nice compromise. I think we would need to make the usage of this config clear in the doc string though.